src/Plugin/resource/DataProvider/DataProviderEntity.php

```diff
@@ -822,11 +829,13 @@ protected function isValidEntity($op, $entity_id) {
         // Just return FALSE, without an exception, for example when a list of
         // entities is requested, and we don't want to fail all the list because
         // of a single item without access.
-
         // Add the inaccessible item to the metadata to fix the record count in
         // the formatter.
         $inaccessible_records = $this->getMetadata()->get('inaccessible_records');
-        $inaccessible_records[] = $entity_id;
+        $inaccessible_records[] = array(
```

Comment:
I see your point. I believe that we need a setting controlling this behavior:

  - [ ] Hide access denied (turns 403 into 404 when selecting entities with access denied)

> I think that when visiting a private entity - one should get a 404 exception (or make it configurable)

I like the idea of making it configurable. I believe that you can do something similar with Drupal pages. We would have a new `UnaccessibleRecordException` and `UnaccessibleRecordException::getCode()` will read from a variable.

> In listing silently ignore the private entities, and don't add their info in the metadata.

I'd make this configurable too, since in cases were the info disclosure is not a concern that is very helpful data. It tells me: «Hey, I know I told you these entities were there but now they are gone all of the sudden. They are gone b/c you don't have access to them.»